libcoap/src/context.rs

```diff
@@ -232,7 +233,7 @@ impl CoapContext<'_> {
     pub fn add_endpoint_udp(&mut self, addr: SocketAddr) -> Result<(), EndpointCreationError> {
         // SAFETY: Because we never return an owned reference to the endpoint, it cannot outlive the
         // context it is bound to (i.e. this one).
-        let endpoint = unsafe { CoapUdpEndpoint::new(self, addr)? }.into();
+        let endpoint = CoapEndpoint::new_udp_endpoint(self, addr)?.into();
```

Comment:
```suggestion
        let endpoint = CoapEndpoint::new_udp_endpoint(self, addr)?;
```

Reply:
Fixed. No need for cast now that there is only one structure.